lib/cylc/rundb.py

```diff
@@ -694,47 +694,37 @@ def select_task_events_for_timing(self):
         """
         q = """
             SELECT
-                e.name,
-                e.cycle,
-                tj.user_at_host,
-                tj.batch_sys_name,
-                MAX(CASE WHEN e.event = '%(submitted)s' THEN e.time END),
-                MAX(CASE WHEN e.event = '%(started)s'   THEN e.time END),
-                MAX(CASE WHEN e.event = '%(succeeded)s' THEN e.time END)
+                %(task_jobs)s.name,
+                %(task_jobs)s.cycle,
+                %(task_jobs)s.user_at_host,
+                %(task_jobs)s.batch_sys_name,
+                %(task_jobs)s.time_submit,
+                %(task_jobs)s.time_run,
+                %(task_jobs)s.time_run_exit
             FROM
-                %(task_events)s as e
+                %(task_jobs)s
                 INNER JOIN
                 (
                     SELECT
-                        s.name AS name,
-                        s.cycle AS cycle,
-                        MAX(s.submit_num) AS max_retry
+                        name,
+                        cycle,
+                        MAX(submit_num) AS max_retry
                     FROM
-                        %(task_events)s as s
+                        %(task_events)s
                     WHERE
-                        s.event='%(succeeded)s'
+                        event = '%(succeeded)s'
```

Comment:
Sorry, I should also have mentioned that `task_jobs.run_status == 0` is probably a better test for succeeded.